### ISSUE TITLE
[Tree] A11y audit

### ISSUE THREAD

**Issue by DanielleRameau** [sl-design-system/components#2519]
### Provide a general summary of the issue here

Test the tree component on a11y, create new issues if there are any findings.

### 👤 Your name

DR

**Comment by KatarzynaWM** [comment]
- the intend lines have too low contrast ratio in both light and dark mode (1,4:1/1,5:1)

<img width="226" height="220" alt="Image" src="https://github.com/user-attachments/assets/7c10fe9f-2244-4e9b-9d04-2ea8d12a8ca9" />

**Comment by KatarzynaWM** [comment]
In Multi select (https://storybook.sanomalearning.design/?path=/story/navigation-tree--multi-select) and Multi select with Badges (https://storybook.sanomalearning.design/?path=/story/navigation-tree--multi-select-with-badges) versions the built in accessibility checker shows problems with wrongly used aria attributes, namely "aria-checked"; The problem here is, that according to documentation, "aria-checked" is not mentioned as to be used with role="row" (https://w3c.github.io/aria/#aria-checked);

Another thing is, that on <input type="checbox"> there is "aria-checked" as well, when it is not needed, as it has built in semantics and doesn't require ARIA (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-checked)

**Comment by KatarzynaWM** [comment]
In Single Select variant (https://storybook.sanomalearning.design/?path=/story/navigation-tree--single-select) color indicators for selected item have to low contrast ratio in both light and dark mode (1,1:1/1,3:1); Should be 3:1

<img width="911" height="145" alt="Image" src="https://github.com/user-attachments/assets/9fd640c5-797e-4f3a-813c-09e3a23ab355" />

**Comment by KatarzynaWM** [comment]
In Single Select, Multi select and Multi select with Badges buttons "toggle selected" and "toggle descendants" are not working as expected. As I understand "toggle selected": 

- Applies only to the item you currently have selected.
- If it’s expanded, it will collapse. If it’s collapsed, it will expand.
- It does not affect any children beyond what’s directly under that selected item.

And "toggle descendants": Applies not just to the selected item, but also to all of its children (descendants).

But the actual behavior in unpredictable. Sometimes it works as intended, sometimes not. For example: on the screenshot below result of clicking "toggle descendants" (the "src" folder has not toggled"):

<img width="993" height="488" alt="Image" src="https://github.com/user-attachments/assets/91c6e045-5514-49e8-8d7a-7df3a3c7d2ab" />

**Comment by KatarzynaWM** [comment]
Elements on level a (no matter the position) are always announced “1 of x level 1” by a screen reader – the real position is not announced.

<img width="982" height="586" alt="Image" src="https://github.com/user-attachments/assets/00933720-bfd4-4889-b386-83ab3e56cc3c" />

**Comment by KatarzynaWM** [comment]
Question: why` role="treegrid"` was used instead of `role="tree"`?

**Comment by KatarzynaWM** [comment]
Question:
According to the ARIA tree view pattern (https://www.w3.org/WAI/ARIA/apg/patterns/treeview/), activating a node (e.g., by clicking or pressing Enter) is typically expected to select the node or trigger an associated action. In most of the current variants, clicking or pressing Enter on a node doesn’t seem to have any effect. Could you share the reasoning behind this behavior?

**Comment by KatarzynaWM** [comment]
Variants with action buttons: when you press "edit" or "remove" buttons - nothing happens.

**Comment by KatarzynaWM** [comment]
What is the use case of “badge”? They are now very difficult to reach with a screen reader.

<img width="1535" height="226" alt="Image" src="https://github.com/user-attachments/assets/6012248f-4145-44ca-969f-b8f15305bd55" />
